lib/client.js

```diff
@@ -228,7 +238,7 @@ class Client extends DispatcherBase {
     this[kMaxRedirections] = maxRedirections
     this[kMaxRequests] = maxRequestsPerClient
     this[kClosedResolve] = null
-    this[kMaxResponseSize] = maxResponseSize > -1 ? maxResponseSize : -1
+    this[kMaxResponseSize] = (typeof maxResponseSize === 'number' && maxResponseSize > -1) ? maxResponseSize : -1
```

Comment:
why this change?

Reply:
Ah, I added `// @ts-check` at the first-line to confirm that the types work.

It showed me the following error on that line 231:

```
'maxResponseSize' is possibly 'undefined'.
```

![image](https://user-images.githubusercontent.com/60829711/221038348-93b899ee-525c-4b8c-b45e-b17460e9dd37.png)

Should I keep it or revert it? `undefined > -1` is `false` anyways.

Reply:
reverted it